lib/paperclip/io_adapters/http_url_proxy_adapter.rb

```diff
@@ -4,9 +4,9 @@ class HttpUrlProxyAdapter < UriAdapter
     REGEXP = /\Ahttps?:\/\//
 
     def initialize(target, options = {})
-      super(URI(URI.escape(target)), options)
+      escaped = URI.escape(target)
+      super(URI(target == URI.unescape(target) ? escaped : target), options)
```

Comment:
It'd be great to merge this. I was about to create an issue in Paperclip just before I saw this. 

Here's the story, if your URI is already `URI.escape`(d), the current code will return bad, double escaped URI. I use a similar method of testing whether the URI is already escaped or not in one of the production apps.